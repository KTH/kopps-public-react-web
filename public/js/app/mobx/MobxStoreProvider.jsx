/* eslint no-use-before-define: ["error", "nofunc"] */

import React from 'react'

import { createMobxManager } from './manager'

export default MobxStoreProvider

/**
 * React-component which prepares a MobX-store
 * so that its data can be used with the useStore() hook
 * of any other component embedded into this Provider.
 *
 * @param {object} props
 * @param {function} props.initCallback
 * @param {string} [props.storeId]
 * @param {object} props.children
 *
 */
function MobxStoreProvider(props) {
  const { initCallback, storeId, children } = props

  const manager = createMobxManager(storeId || 'default', initCallback)

  const { Provider: ReactContextProvider, value: mobxStore } = manager.getProviderComponentAndValue()

  return <ReactContextProvider value={mobxStore}>{children}</ReactContextProvider>
}
