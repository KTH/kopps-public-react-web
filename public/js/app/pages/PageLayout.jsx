import React from 'react'
import { Col, Row } from 'reactstrap'
import MainMenu from '../components/MainMenu'

function MainContent({ children }) {
  return <main id="mainContent">{children}</main>
}

function PageLayout({ children }) {
  return (
    <Row style={{ padding: '30px 0' }}>
      <MainMenu />
      <Col>
        <MainContent>{children}</MainContent>
      </Col>
    </Row>
  )
}

export default PageLayout
