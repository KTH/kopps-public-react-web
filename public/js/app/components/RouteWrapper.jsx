/* eslint-disable react/jsx-props-no-spreading */
import React from 'react'
import { Route } from 'react-router-dom'
import PropTypes from 'prop-types'
import { MobxStoreProvider } from '../mobx'

function RouteWrapper({
  createBreadcrumbs,
  component: Component,
  layout: Layout,
  applicationStore,
  createMenuData,
  ...rest
}) {
  const menuData = createMenuData(applicationStore)
  const breadcrumbs = createBreadcrumbs(applicationStore)
  return (
    <MobxStoreProvider initCallback={() => applicationStore}>
      <Route
        {...rest}
        render={() => (
          <Layout breadcrumbs={breadcrumbs} menuData={menuData}>
            <Component />
          </Layout>
        )}
      />
    </MobxStoreProvider>
  )
}

RouteWrapper.propTypes = {
  createBreadcrumbs: PropTypes.func,
  component: PropTypes.oneOfType([PropTypes.node, PropTypes.func, PropTypes.elementType]).isRequired,
  layout: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
  createMenuData: PropTypes.func.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  applicationStore: PropTypes.object.isRequired,
}

RouteWrapper.defaultProps = {
  createBreadcrumbs: () => ({ include: 'student', items: [] }),
}

export default RouteWrapper
