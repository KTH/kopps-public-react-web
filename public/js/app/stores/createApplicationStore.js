/* eslint no-use-before-define: ["error", "nofunc"] */

// @ts-check

// eslint-disable-next-line no-unused-vars
import { observable } from 'mobx'
import commonStore from './commonStore'
import curriculumStore from './curriculumStore'
import studyProgrammeStore from './studyProgrammeStore'
import searchCoursesStore from './searchCoursesStore'
import appendix1Store from './appendix1Store'
import appendix2Store from './appendix2Store'
import literatureStore from './literatureStore'

export default createApplicationStore

const store = {
  /**
   * @property {string} message
   */
  message: 'Hallo',
  /**
   * @method
   * @param {string} text
   */
  setMessage,
  /**
   * @property {string} lastAdmissionTerm
   */
  lastAdmissionTerm: '',
  /**
   * @method
   * @param {string} lastAdmissionTerm
   */
  setLastAdmissionTerm,
  /**
   * @method
   * @param {map<string, {}>} programmes
   */
  setProgrammes,

  /**
   * @property {map<string, {}>} programmes
   */
  programmes: [],

  /**
   * @method
   * @param {[]} currentSchoolsWithDepartments
   */
  setCurrentSchoolsWithDepartments,

  /**
   * @property {[]} currentSchoolsWithDepartments
   */
  currentSchoolsWithDepartments: [],

  /**
   * @method
   * @param {[]} deprecatedSchoolsWithDepartments
   */
  setDeprecatedSchoolsWithDepartments,

  /**
   * @property {[]} deprecatedSchoolsWithDepartments
   */
  deprecatedSchoolsWithDepartments: [],
  /**
   * @property {string} departmentName
   */
  departmentName: '',
  /**
   * @method
   * @param {string} departmentName
   */
  setDepartmentName,

  /**
   * @property {[]} departmentCourses
   */
  departmentCourses: [],
  /**
   * @method
   * @param {[]} departmentCourses
   */
  setDepartmentCourses,
  /**
   * @property {string} programmeName
   */
  programmeName: '',
  /**
   * @method
   * @param {string} programmeName
   */
  setProgrammeName,
  /**
   * @property {string} programmeCode
   */
  programmeCode: '',
  /**
   * @method
   * @param {string} programmeCode
   */
  setProgrammeCode,
  /**
   * @property {[]} programmeTerms
   */
  programmeTerms: [],
  /**
   * @method
   * @param {[]} programmeTerms
   */
  setProgrammeTerms,
  /**
   * @property {number} lengthInStudyYears
   */
  lengthInStudyYears: 0,
  /**
   * @method
   * @param {number} lengthInStudyYears
   */
  setLengthInStudyYears,
  /**
   * @property {string} thisHostBaseUrl
   */
  thisHostBaseUrl: null,
}

function createApplicationStore(storeId) {
  switch (storeId) {
    case 'curriculum':
      return {
        ...commonStore,
        ...curriculumStore,
      }
    case 'searchCourses':
      return {
        ...commonStore,
        ...searchCoursesStore,
      }
    case 'objectives':
    case 'extent':
    case 'eligibility':
    case 'implementation':
      return {
        ...commonStore,
        ...studyProgrammeStore,
      }
    case 'appendix1':
      return {
        ...commonStore,
        ...studyProgrammeStore,
        ...appendix1Store,
      }
    case 'appendix2':
      return {
        ...commonStore,
        ...studyProgrammeStore,
        ...appendix2Store,
      }
    case 'literatureList':
      return {
        ...commonStore,
        ...literatureStore,
      }
    default:
      return {
        ...commonStore,
        ...store,
      }
  }
}

function setMessage(text = 'Happy coding!! :)') {
  this.message = text
}

function setProgrammes(programmes) {
  this.programmes = [...programmes]
}

function setCurrentSchoolsWithDepartments(currentSchoolsWithDepartments) {
  this.currentSchoolsWithDepartments = currentSchoolsWithDepartments
}

function setDeprecatedSchoolsWithDepartments(deprecatedSchoolsWithDepartments) {
  this.deprecatedSchoolsWithDepartments = deprecatedSchoolsWithDepartments
}

function setDepartmentName(departmentName) {
  this.departmentName = departmentName
}

function setDepartmentCourses(departmentCourses) {
  this.departmentCourses = departmentCourses
}

function setLastAdmissionTerm(lastAdmissionTerm) {
  this.lastAdmissionTerm = lastAdmissionTerm
}

function setProgrammeName(programmeName) {
  this.programmeName = programmeName
}

function setProgrammeCode(programmeCode) {
  this.programmeCode = programmeCode
}

function setProgrammeTerms(programmeTerms) {
  this.programmeTerms = programmeTerms
}

function setLengthInStudyYears(lengthInStudyYears) {
  this.lengthInStudyYears = lengthInStudyYears
}
