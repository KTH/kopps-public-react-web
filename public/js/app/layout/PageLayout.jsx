import React, { useEffect } from 'react'
import ReactDOM from 'react-dom'
import { Col, Row } from 'reactstrap'
import { Breadcrumbs } from '@kth/kth-reactstrap/dist/components/utbildningsinfo'

import { useStore } from '../mobx'
import MainMenu from '../components/MainMenu'

function renderBreadcrumbsIntoKthHeader(language) {
  const breadcrumbContainer = document.getElementById('breadcrumbs-header')
  if (breadcrumbContainer) {
    ReactDOM.render(<Breadcrumbs include="directory" language={language} />, breadcrumbContainer)
  }
}

function MainContent({ children }) {
  return (
    <Col>
      <main id="mainContent">{children}</main>
    </Col>
  )
}

function PageLayout({ children }) {
  const { language } = useStore()
  useEffect(() => renderBreadcrumbsIntoKthHeader(language))
  return (
    // Container in publicLayout.handlebars – begin
    <Row>
      <MainMenu />
      <MainContent>{children}</MainContent>
    </Row>
    // Container – end
  )
}

export default PageLayout
