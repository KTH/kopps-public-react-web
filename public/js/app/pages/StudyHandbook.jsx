import React from 'react'
import { Col, Row } from 'reactstrap'
import { PageHeading } from '@kth/kth-reactstrap/dist/components/studinfo'

import Article from '../components/Article'

import FooterContent from '../components/FooterContent'

import { useStore } from '../mobx'
import i18n from '../../../../i18n'

function StudyHandbook() {
  const { languageIndex } = useStore()
  const { pageHeading, shbLink, content } = i18n.messages[languageIndex].shb

  return (
    <>
      <Row>
        <Col>
          <PageHeading>{pageHeading}</PageHeading>
        </Col>
      </Row>
      <Row>
        <Col>
          <Article>
            <p>{content}</p>
            <p>
              <a href="https://intra.kth.se/utbildning/systemstod/kopps/shb">{shbLink}</a>
            </p>
          </Article>
        </Col>
      </Row>
      <Row>
        <Col>
          <FooterContent />
        </Col>
      </Row>
    </>
  )
}

export default StudyHandbook
