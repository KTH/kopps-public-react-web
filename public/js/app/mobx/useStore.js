/* eslint no-use-before-define: ["error", "nofunc"] */

import createApplicationStore from '../stores/createApplicationStore'
import { getMobxManager } from './manager'

export default useStore

/**
 * Hook which can be used by any React component
 * within the related MobxStoreProvider.
 *
 * @param {string} [storeId]
 *
 *      Current state of application store
 */
function useStore(storeId = 'default') {
  const manager = getMobxManager(storeId)

  const isTestEnvironmentWithoutProviderComponent = manager == null
  if (isTestEnvironmentWithoutProviderComponent) {
    const dummyStoreForTesting = createApplicationStore()
    return dummyStoreForTesting
  }

  return manager.getContextHook()
}
