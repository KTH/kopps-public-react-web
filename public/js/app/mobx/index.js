/* eslint no-use-before-define: ["error", "nofunc"] */

import MobxStoreProvider from './MobxStoreProvider'
import useStore from './useStore'
import { compressStoreIntoJavascriptCode, uncompressStoreInPlaceFromDocument } from './compress'
// import { compressStoreIntoJavascriptCode, uncompressStoreInPlaceFromDocument } from './compressWithPako'

export { MobxStoreProvider, useStore, compressStoreIntoJavascriptCode, uncompressStoreInPlaceFromDocument }
