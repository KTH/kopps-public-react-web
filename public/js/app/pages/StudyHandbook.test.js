/* eslint-disable react/prop-types */

import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { StaticRouter } from 'react-router'
import { MobxStoreProvider } from '../mobx'

import StudyHandbook from './StudyHandbook'
import RouteWrapper from '../components/RouteWrapper'
import PageLayout from '../layout/PageLayout'

import createApplicationStore from '../stores/createApplicationStore'
import getMenuData from '../config/menuData'
const applicationStore = createApplicationStore()

const { getAllByRole, getAllByTestId, getAllByText, getByTestId, getByText } = screen

const testProxyPath = 'localhost://kopps-public'
const proxyPrefixPath = { uri: testProxyPath }

const WrapperStudyHandbook = ({ lang, ...rest }) => {
  applicationStore.setLanguage(lang)
  applicationStore.setBrowserConfig({ proxyPrefixPath })

  const updatedApplicationStore = {
    ...applicationStore,
  }
  return (
    <StaticRouter>
      <RouteWrapper
        exact
        path="/student/program/shb"
        component={StudyHandbook}
        applicationStore={updatedApplicationStore}
        createBreadcrumbs={() => ({ include: 'directory' })}
        layout={PageLayout}
        createMenuData={store => ({ selectedId: 'shb', ...getMenuData(store) })}
        {...rest}
      />
    </StaticRouter>
  )
}

const StudyHandbookWithLayout = ({ lang }) => {
  applicationStore.setLanguage(lang)
  applicationStore.setBrowserConfig({ proxyPrefixPath })
  const menuData = getMenuData(applicationStore)

  const updatedApplicationStore = {
    ...applicationStore,
  }
  return (
    <StaticRouter>
      <MobxStoreProvider initCallback={() => updatedApplicationStore}>
        <PageLayout breadcrumbs={{ include: 'directory' }} menuData={{ selectedId: 'shb', ...menuData }}>
          <StudyHandbook />
        </PageLayout>
      </MobxStoreProvider>
    </StaticRouter>
  )
}

describe('Render component StudyHandbook within RouterWrapper', () => {
  test('simple render', done => {
    render(<WrapperStudyHandbook lang="en" />)
    done()
  })
})

describe('Check if getMenuData is correctly fetching menuData', () => {
  test('simple render', done => {
    render(<WrapperStudyHandbook lang="en" />)
    done()
  })
})

describe('Render component StudyHandbook within Layout', () => {
  test('get page header in English', () => {
    render(<StudyHandbookWithLayout lang="en" />)
    const h1Header = screen.getByRole('heading', { level: 1 })
    expect(h1Header).toHaveTextContent('Study Handbook 00/01 to 07/08')
  })

  test('get page header in Swedish', () => {
    render(<StudyHandbookWithLayout lang="sv" />)
    const h1Header = screen.getByRole('heading', { level: 1 })
    expect(h1Header).toHaveTextContent('Studiehandboken 00/01 tom 07/08')
  })

  test('match to snapshot in English', done => {
    const { asFragment } = render(<StudyHandbookWithLayout lang="en" />)
    expect(asFragment()).toMatchSnapshot()

    done()
  })

  test('match to snapshot in Swedish', done => {
    const { asFragment } = render(<StudyHandbookWithLayout lang="sv" />)
    expect(asFragment()).toMatchSnapshot()

    done()
  })
})

describe('Render component StudyHandbook and check its menu, content and links', () => {
  test('check all links on the page in English', () => {
    render(<StudyHandbookWithLayout lang="en" />)
    const links = screen.getAllByRole('link')
    expect(links.length).toBe(8)
    expect(links[0]).toHaveTextContent('Student at KTH')
    expect(links[0].href).toStrictEqual('http://localhost/student/?l=en')

    expect(links[1]).toHaveTextContent('Courses Part of Programme')
    expect(links[1].href).toStrictEqual('localhost://kopps-public/student/kurser/kurser-inom-program')

    expect(links[2]).toHaveTextContent('Search course')
    expect(links[2].href).toStrictEqual('localhost://kopps-public/student/kurser/sokkurs')

    expect(links[3]).toHaveTextContent('Courses by school')
    expect(links[3].href).toStrictEqual('localhost://kopps-public/student/kurser/org')

    expect(links[4]).toHaveTextContent('Study Handbook 00/01 to 07/08') // menu link
    expect(links[4].href).toStrictEqual('localhost://kopps-public/student/program/shb')

    expect(links[5]).toHaveTextContent('Study Handbook 00/01 to 07/08') // content link
    expect(links[5].href).toStrictEqual('https://intra.kth.se/utbildning/utbildningsadministr/kopps/shb')

    expect(links[6]).toHaveTextContent('Central study counseling')
    expect(links[6].href).toStrictEqual('https://www.kth.se/studycounselling')

    expect(links[7]).toHaveTextContent('kopps@kth.se')
    expect(links[7].href).toStrictEqual('mailto:kopps@kth.se')
  })

  test('check all links on the page in Swedish', () => {
    render(<StudyHandbookWithLayout lang="sv" />)
    const links = screen.getAllByRole('link')
    expect(links.length).toBe(8)
    expect(links[0]).toHaveTextContent('Student på KTH')
    expect(links[0].href).toStrictEqual('http://localhost/student/')

    expect(links[1]).toHaveTextContent('Kurser inom program')
    expect(links[1].href).toStrictEqual('localhost://kopps-public/student/kurser/kurser-inom-program')

    expect(links[2]).toHaveTextContent('Sök kurs')
    expect(links[2].href).toStrictEqual('localhost://kopps-public/student/kurser/sokkurs')

    expect(links[3]).toHaveTextContent('Kurser per skola')
    expect(links[3].href).toStrictEqual('localhost://kopps-public/student/kurser/org')

    expect(links[4]).toHaveTextContent('Studiehandboken 00/01 tom 07/08') // menu link
    expect(links[4].href).toStrictEqual('localhost://kopps-public/student/program/shb')

    expect(links[5]).toHaveTextContent('Studiehandboken 00/01 tom 07/08') // content link
    expect(links[5].href).toStrictEqual('https://intra.kth.se/utbildning/utbildningsadministr/kopps/shb')

    expect(links[6]).toHaveTextContent('Central studievägledning')
    expect(links[6].href).toStrictEqual('https://www.kth.se/studycounselling')

    expect(links[7]).toHaveTextContent('kopps@kth.se')
    expect(links[7].href).toStrictEqual('mailto:kopps@kth.se')
  })

  test('get page content in English', () => {
    render(<StudyHandbookWithLayout lang="en" />)
    const content = screen.getByText(
      'Until the study year 2007/2008 the study handbook contained program and course information.'
    )
    expect(content).toBeInTheDocument()
  })

  test('get page content in Swedish', () => {
    render(<StudyHandbookWithLayout lang="sv" />)
    const content = screen.getByText(
      'Fram till läsåret 2007/2008 fanns information om program och kurser i studiehandboken.'
    )
    expect(content).toBeInTheDocument()
  })
})
