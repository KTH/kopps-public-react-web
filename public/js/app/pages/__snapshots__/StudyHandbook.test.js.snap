// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component StudyHandbook within Layout match to snapshot in English 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <nav
      aria-labelledby="kth-local-navigation-title--mobile"
      class="kth-local-navigation--mobile"
    >
      <button
        class="kth-button menu"
        id="kth-local-navigation-title--mobile"
      >
        <span>
          Course and programme directory
        </span>
      </button>
      <dialog
        class="kth-mobile-menu left"
      >
        <div
          class="kth-mobile-menu__navigation"
        >
          <button
            class="kth-icon-button close"
          >
            <span
              class="kth-visually-hidden"
            >
              Close
            </span>
          </button>
        </div>
        <div
          class="mobile-menu__content"
        >
          <a
            class="kth-button back"
            href="/student/?l=en"
          >
            Student at KTH
          </a>
          <h2>
            Course and programme directory
          </h2>
          <ul>
            <li>
              <a
                href="/student/kurser/kurser-inom-program"
              >
                Programme Syllabuses
              </a>
            </li>
            <li>
              <a
                href="/student/kurser/sokkurs"
              >
                Search courses
              </a>
            </li>
            <li>
              <a
                href="/student/kurser/org"
              >
                Courses by school
              </a>
            </li>
            <li>
              <a
                aria-current="page"
                href="/student/program/shb"
              >
                Studies before 07/08
              </a>
            </li>
          </ul>
        </div>
      </dialog>
    </nav>
    <nav
      aria-labelledby="local-navigation-title"
      class="kth-local-navigation col"
      id="mainMenu"
    >
      <a
        class="kth-button back"
        href="/student/?l=en"
      >
        Student at KTH
      </a>
      <h2
        id="local-navigation-title"
      >
        Course and programme directory
      </h2>
      <ul>
        <li>
          <a
            href="/student/kurser/kurser-inom-program"
          >
            Programme Syllabuses
          </a>
        </li>
        <li>
          <a
            href="/student/kurser/sokkurs"
          >
            Search courses
          </a>
        </li>
        <li>
          <a
            href="/student/kurser/org"
          >
            Courses by school
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            href="/student/program/shb"
          >
            Studies before 07/08
          </a>
        </li>
      </ul>
    </nav>
    <main
      class="col"
      id="mainContent"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <header
            role="presentation"
          >
            <h1>
              Studies before 07/08
            </h1>
          </header>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <article
            class="article article--no-start-margin "
          >
            <p>
              Until the academic year 2007/2008, information on programmes and courses was available in the study handbook.
            </p>
            <p>
              <a
                href="https://intra.kth.se/utbildning/systemstod/kopps/shb"
              >
                Study Handbook 00/01 to 07/08
              </a>
            </p>
          </article>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <footer
            class="border-top mt-4 pt-1"
            id="articleFooter"
          >
            <address>
              Contact for questions regarding education at KTH: 
              <a
                href="https://www.kth.se/en/studies/master/general-study-counselling-1.621634"
              >
                Central study counseling
              </a>
              <br />
              Contact regarding technical matters on this page: 
              <a
                href="mailto:kopps@kth.se"
              >
                kopps@kth.se
              </a>
            </address>
          </footer>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;

exports[`Render component StudyHandbook within Layout match to snapshot in Swedish 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <nav
      aria-labelledby="kth-local-navigation-title--mobile"
      class="kth-local-navigation--mobile"
    >
      <button
        class="kth-button menu"
        id="kth-local-navigation-title--mobile"
      >
        <span>
          Kurs- och programkatalogen
        </span>
      </button>
      <dialog
        class="kth-mobile-menu left"
      >
        <div
          class="kth-mobile-menu__navigation"
        >
          <button
            class="kth-icon-button close"
          >
            <span
              class="kth-visually-hidden"
            >
              Close
            </span>
          </button>
        </div>
        <div
          class="mobile-menu__content"
        >
          <a
            class="kth-button back"
            href="/student/"
          >
            Student på KTH
          </a>
          <h2>
            Kurs- och programkatalogen
          </h2>
          <ul>
            <li>
              <a
                href="/student/kurser/kurser-inom-program"
              >
                Utbildningsplaner
              </a>
            </li>
            <li>
              <a
                href="/student/kurser/sokkurs"
              >
                Sök kurser
              </a>
            </li>
            <li>
              <a
                href="/student/kurser/org"
              >
                Kurser per skola
              </a>
            </li>
            <li>
              <a
                aria-current="page"
                href="/student/program/shb"
              >
                Studier före 07/08
              </a>
            </li>
          </ul>
        </div>
      </dialog>
    </nav>
    <nav
      aria-labelledby="local-navigation-title"
      class="kth-local-navigation col"
      id="mainMenu"
    >
      <a
        class="kth-button back"
        href="/student/"
      >
        Student på KTH
      </a>
      <h2
        id="local-navigation-title"
      >
        Kurs- och programkatalogen
      </h2>
      <ul>
        <li>
          <a
            href="/student/kurser/kurser-inom-program"
          >
            Utbildningsplaner
          </a>
        </li>
        <li>
          <a
            href="/student/kurser/sokkurs"
          >
            Sök kurser
          </a>
        </li>
        <li>
          <a
            href="/student/kurser/org"
          >
            Kurser per skola
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            href="/student/program/shb"
          >
            Studier före 07/08
          </a>
        </li>
      </ul>
    </nav>
    <main
      class="col"
      id="mainContent"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <header
            role="presentation"
          >
            <h1>
              Studier före 07/08
            </h1>
          </header>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <article
            class="article article--no-start-margin "
          >
            <p>
              Fram till läsåret 2007/2008 fanns information om program och kurser i studiehandboken.
            </p>
            <p>
              <a
                href="https://intra.kth.se/utbildning/systemstod/kopps/shb"
              >
                Studiehandboken 00/01 tom 07/08
              </a>
            </p>
          </article>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <footer
            class="border-top mt-4 pt-1"
            id="articleFooter"
          >
            <address>
              Kontakt för frågor om utbildning på KTH: 
              <a
                href="https://www.kth.se/studievagledning"
              >
                Central studievägledning
              </a>
              <br />
              Teknisk kontakt om denna sida: 
              <a
                href="mailto:kopps@kth.se"
              >
                kopps@kth.se
              </a>
            </address>
          </footer>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
